### PR TITLE
SearchDenseVectors pass the ThreadPoolExecutor to IndexSearcher for parallelizing searches

### PR DESCRIPTION
**Pull request by tteofili**
Make it possible to leverage parallel index search in `SearchDenseVectors` by passing the `Executor` to `IndexSearcher` via `-parallel` param.

### CODE REVIEW

**Comment by lintool** [comment]
hey @tteofili let's completely redo the parallelism options into something that actually makes sense? the `-threads` and `-parallelism' options is technical debt and confusing, copied over from `SearchCollection`. Since we're starting with a clean slate, we should just implement it "right".

**Comment by lintool** [comment]
So _Anserini_ would be able to run _n_ queries in parallel, and for each query you'd have a threadpool of _m_ threads? So that's a total of n * m threads?

For modern hardware, with so many cores, that makes sense.

**Comment by tteofili** [comment]
yes that is the idea. That way we can more easily evaluate the performance speedup provided by parallelizing each part.

**Comment by tteofili** [comment]
moved the `-parallel` option to `-threadsPerSearch`.

**Comment by lintool** [comment]
> moved the `-parallel` option to `-threadsPerSearch`.

How about `-threadsPerQuery`?

**Comment by tteofili** [comment]
> How about `-threadsPerQuery`?

👍

File: src/main/java/io/anserini/search/SearchDenseVectors.java [hunk 1]
```diff
@@ -341,22 +344,43 @@ public <K> void runTopics() throws IOException {
 
 
     LOG.info("============ Launching Search Threads ============");
-
+    ThreadPoolExecutor searcherExecutor;
+    if (args.threadsPerQuery > 1) {
+      searcherExecutor = (ThreadPoolExecutor) Executors.newFixedThreadPool(args.threadsPerQuery);
+    } else {
+      searcherExecutor = null;
```
Comment (lintool) [inline src/main/java/io/anserini/search/SearchDenseVectors.java:351]: if you set `ThreadPoolExecutor searcherExecutor = null` above, you won't need this branch?

**Review by lintool — APPROVED** [review]
Minor suggestions...

**Pull request closed without merging**